addons/docs/src/frameworks/vue/sourceDecorator.ts

```diff
@@ -30,47 +30,45 @@ export const sourceDecorator = (storyFn: any, context: StoryContext) => {
     return story;
   }
 
-  try {
-    // Creating a Vue instance each time is very costly. But we need to do it
-    // in order to access VNode, otherwise vm.$vnode will be undefined.
-    // Also, I couldn't see any notable difference from the implementation with
-    // per-story-cache.
-    // But if there is a more performant way, we should replace it with that ASAP.
-    const vm = new Vue({
-      data() {
-        return {
-          STORYBOOK_VALUES: context.args,
-        };
-      },
-      render(h) {
-        return h(story);
-      },
-    }).$mount();
-
-    const channel = addons.getChannel();
-
-    const storyComponent = getStoryComponent(story.options.STORYBOOK_WRAPS);
-
-    const storyNode = lookupStoryInstance(vm, storyComponent);
-
-    const code = vnodeToString(storyNode._vnode);
-
-    channel.emit(
-      SNIPPET_RENDERED,
-      (context || {}).id,
-      prettier.format(`<template>${code}</template>`, {
-        parser: 'vue',
-        plugins: [prettierHtml],
-        // Because the parsed vnode missing spaces right before/after the surround tag,
-        // we always get weird wrapped code without this option.
-        htmlWhitespaceSensitivity: 'ignore',
-      })
-    );
-  } catch (e) {
-    logger.warn(`Failed to generate dynamic story source: ${e}`);
-  }
+  const channel = addons.getChannel();
+
+  const storyComponent = getStoryComponent(story.options.STORYBOOK_WRAPS);
 
-  return story;
+  return {
+    components: {
+      Story: story,
+    },
+    // We need to wait until the wrapper component to be mounted so Vue runtime
+    // struct VNode tree. We get `this._vnode == null` if switch to `created`
+    // lifecycle hook.
+    mounted() {
+      // Theoretically this does not happens but we need to check it.
+      if (!this._vnode) {
+        return;
+      }
+
+      try {
+        const storyNode = lookupStoryInstance(this, storyComponent);
+
+        const code = vnodeToString(storyNode._vnode);
+
+        channel.emit(
+          SNIPPET_RENDERED,
+          (context || {}).id,
+          prettier.format(`<template>${code}</template>`, {
+            parser: 'vue',
+            plugins: [prettierHtml],
+            // Because the parsed vnode missing spaces right before/after the surround tag,
+            // we always get weird wrapped code without this option.
+            htmlWhitespaceSensitivity: 'ignore',
+          })
+        );
+      } catch (e) {
+        logger.warn(`Failed to generate dynamic story source: ${e}`);
+      }
+    },
+    template: '<story />',
```

Comment:
Should I write in this form?

```diff
 return {
- components: {
-    Story: story,
-  },
   mounted() {
     // ...
   },
-  template: '<story />',
+  render(h) { return h(story) },
 }
```

Reply:
What's the difference?

Reply:
AFAIK there is no difference. Just coding style :)

Reply:
I'm fine with the current then. 👍